plugins/hypervisors/hyperv/src/com/cloud/hypervisor/hyperv/resource/HypervDirectConnectResource.java

```diff
@@ -636,7 +637,7 @@ public ExecutionResult executeInVR(final String routerIP, final String script, f
 
         try {
             result = SshHelper.sshExecute(routerIP, DEFAULT_DOMR_SSHPORT, "root", getSystemVMKeyFile(), null, "/opt/cloud/bin/" + script + " " + args,
-                    60000, 60000, timeout * 1000);
+                    60000, 60000, (int)timeout.getMillis());
```

Comment:
Please consider defining an override of the ``SshHelper.sshExecute`` method that accepts a ``Duration`` in order to encapsulate this conversion.  We should also consider deprecating the version that accepts ``int``.